packages/aws-cdk/lib/toolkit/cli-io-host.ts

```diff
@@ -133,6 +182,50 @@ export class CliIoHost {
   }
 }
 
+/**
+ * Validates a message code. The code must:
+ * - Be in the format [A-Z]+_[0-2][0-9]{3}
+ * - Have a numeric prefix that matches the message level:
+ *   - Error level codes must begin with 0
+ *   - Warning level codes must begin with 1
+ *   - Info/debug/trace level codes must begin with 2
+ *
+ * @param code The message code to validate
+ * @param level The IoMessageLevel of the message
+ * @returns boolean indicating if the code is valid for the given level
+ */
+export function validateMessageCode(code: string, level?: IoMessageLevel): boolean {
```

Comment:
i would prefer this function return not just a true/false but information on why/what the errors are. usually achieved by returning a list of errors, with `[]` indicating "no error"

Reply:
Eh maybe, either way I'll refactor this into something after we confirm how exactly we want message codes to look. Might just throw errors tbh